From 18944a8039e3d2fc40ed29c2adc687e10e9191c1 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 15 Oct 2017 10:55:35 +0300
Subject: [PATCH 2034/2241] fix: net: mvpp2x: fix priv->bm_pools memory leakage
 during S2RAM

Issue:
New priv->bm_pools pointer allocated during S2RAM resume by mv_pp2x_init
routine and priv->bm_pools wasn't released during S2RAM suspend.
This caused memory leakage.

Fix:
Release priv->bm_pools pointer during S2RAM suspend routine.

Change-Id: I46d5e1c30004e960c5ff93eacd521a2bac545711
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45116
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 3505040..155b029 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -6229,6 +6229,8 @@ static int mvpp2x_suspend(struct device *dev)
 		mv_pp2x_bm_pool_destroy(dev, priv, bm_pool);
 	}
 
+	devm_kfree(&pdev->dev, priv->bm_pools);
+
 	return 0;
 }
 
-- 
2.7.4

